pkg/helm/handlers/handlers.go

```diff
@@ -240,3 +250,27 @@ func (h *helmHandlers) HandleGetReleaseHistory(user *auth.User, w http.ResponseW
 	w.Header().Set("Content-Type", "application/json")
 	w.Write(res)
 }
+
+func (h *helmHandlers) HandleGetRepos(user *auth.User, w http.ResponseWriter, r *http.Request) {
+	conf := h.getActionConfigurations(h.ApiServerHost, "", user.Token, &h.Transport)
+	client, err := h.getDynamicClient(conf)
+	if err != nil {
+		serverutils.SendResponse(w, http.StatusInternalServerError, serverutils.ApiError{Err: fmt.Sprintf("Failed to get k8s dynamic client: %v", err)})
+		return
+	}
+	w.Header().Set("Content-Type", "application/yaml")
+	w.Header().Set("Cache-Control", "private, no-cache")
+
+	repos, err := h.getRepos(client)
+	if err != nil || (repos != nil && len(repos.Entries) == 0) {
+		fmt.Println("Fallback to proxy ", err)
+		helmChartRepoProxy := proxy.NewProxy(h.ChartProxyConfig)
+		http.StripPrefix(
+			proxy.SingleJoiningSlash("", "/api/helm/charts"),
+			http.HandlerFunc(helmChartRepoProxy.ServeHTTP)).ServeHTTP(w, r)
+		return
+	}
```

Comment:
we shoudl handle the default situation in `h.getRepos`, not here, to separate concerns.